app/models/role.rb

```diff
@@ -155,6 +155,24 @@ def set_defaults
     self.active = true if self.new_record?
   end
 
+  def deactivate!
+    self.active = false
+    if self.save!
+      # If no other creator or administrator is attached to the plan
+      # the deactivate all the other roles
+      if Role.where(plan: self.plan, active: true)
+             .with_access_flags(:creator, :administrator).size <= 0
+        [:editor, :commenter, :reviewer].each do |access|
+          Role.where(plan: self.plan, active: true)
+              .with_access_flags(access).update_all(active: false)
+        end
```

Comment:
I think that the .with_access_flags() function is duplicating functionality built into flag_shih_tzu.
https://github.com/pboling/flag_shih_tzu#generated-named-scopes

Also the way we have defined the access fields, all creators will be admins, so we shouldnt need to check for both.

```suggestion
        if Role.where(plan: self.plan, active: true).administrator.size == 0
          Role.where(plan: self.plan, active: true).update_all(active: true)
        end
```